docs/design/2022-06-15-phyical_plan_explain.md

```diff
@@ -0,0 +1,451 @@
+   ```
+
+   ```SQL
+   SELECT * FROM t2 where a<1 or b>'bb'"
+   ```
+
+   ```text
+   == Physical Plan ==
+   *(1) ColumnarToRow
+   +- TiSpark RegionTaskExec{downgradeThreshold=1000000000,downgradeFilter=[[[a@UNSIGNED LONG LESS_THAN 1] OR [b@VARCHAR(255) GREATER_THAN "bb"]]]
+      +- RowToColumnar
+         +- TiKV FetchHandleRDD{[table: t2] IndexScan[Index: primary] , Columns: a@UNSIGNED LONG, b@VARCHAR(255), c@VARCHAR(255), Downgrade Filter: [[a@UNSIGNED LONG LESS_THAN 1] OR [b@VARCHAR(255) GREATER_THAN "bb"]], PushDown Filter: [[a@UNSIGNED LONG LESS_THAN 1] OR [b@VARCHAR(255) GREATER_THAN "bb"]], KeyRange: [([t\200\000\000\000\000\000\003\374_i\200\000\000\000\000\000\000\001\000], [t\200\000\000\000\000\000\003\374_i\200\000\000\000\000\000\000\001\372])], Aggregates: , startTs: 433735624236728323}
+   ```
+
+3. expression(s) that constitutes the scanning range is hard to know
+
+   The expression(s) for building ranges are not shown explicitly, which makes it difficult to know which expression(s) are used to build range. As shown below, for the query expression `a>0` is pushed down, but the pushed-down information is only implicitly given inside the `KeyRange`, which is not convenient for users to understand.
+
+   ```sql
+   CREATE TABLE `t3` (
+     `a` BIGINT(20) NOT NULL,
+     `b` varchar(255) NOT NULL,
+     `c` varchar(255) DEFAULT NULL,
+     PRIMARY KEY (`a`) clustered
+   ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
+   ```
+
+   ```sql
+   SELECT a FROM t3 where a>0
+   ```
+
+   ```text
+   == Physical Plan ==
+   *(1) ColumnarToRow
+   +- TiKV CoprocessorRDD{[table: t3] TableScan, Columns: a@LONG, KeyRange: [([t\200\000\000\000\000\000\005/_r\200\000\000\000\000\000\000\001], [t\200\000\000\000\000\000\005/_s\000\000\000\000\000\000\000\000])], startTs: 434063568626778113}
+   ```
+
+4. confusing naming of the operator
+
+   In TiDB, a process named `Scan` means scanning a piece of data, not aggregating and computing the result of the scan. But in TiSpark, a process named scan means aggregating and computing the results of the scan. This may confuse users who switch from TiDB to TiSpark.
+
+   The selection expression pushed down to TiKV/COP in TiDB is called `Selection`, while the selection expression is called `Pushdown Filter` in TiSpark. This may also confuse users who switch from TiDB to TiSpark.
+
+## Detailed Design
+
+### Change to now output
+
+#### appear a filter that should not appear
+
+To solve the problem of appearing a filter that shouldn't appear, we removed the `Residual Filter`.
+
+#### unclear representation of the execution process
+
+To solve the problem of the execution process for `IndexScan` is divided into two phases is not show, we divide the scan of `IndexScan` into two parts `IndexRangeScan`, `TableRowIDScan`. The scan of `TableScan` is named `TableRangeScan`. The scan of `CoveringIndexScan` is named `IndexRangeScan`. The meanings of these scans are shown below.
```

Comment:
I think this part should move to confusing naming of the operator

Reply:
I think the reason for introducing these scans is to better represent the execution process. So it might be better to put it here.